scala/scala.bzl

```diff
@@ -772,22 +772,35 @@ java_import(
     jars = ["lib/scala-reflect.jar"],
     visibility = ["//visibility:public"],
 )
+
+java_library(
+    name = "transitive_scalatest",
+    exports = ["@scalatest//jar", "@scalactic//jar"],
+    visibility = ["//visibility:public"],
+)
 """
 
 def scala_repositories():
   native.new_http_archive(
     name = "scala",
-    strip_prefix = "scala-2.11.11",
-    sha256 = "12037ca64c68468e717e950f47fc77d5ceae5e74e3bdca56f6d02fd5bfd6900b",
-    url = "https://downloads.lightbend.com/scala/2.11.11/scala-2.11.11.tgz",
+    strip_prefix = "scala-2.12.2",
+    sha256 = "196168b246fcf10e275491c5e58a50ca9eb696da95e49155b3f86f001346a6f5",
+    url = "https://downloads.lightbend.com/scala/2.12.2/scala-2.12.2.tgz",
     build_file_content = SCALA_BUILD_FILE,
   )
 
   # scalatest has macros, note http_jar is invoking ijar
   native.http_jar(
     name = "scalatest",
-    url = "http://mirror.bazel.build/oss.sonatype.org/content/groups/public/org/scalatest/scalatest_2.11/2.2.6/scalatest_2.11-2.2.6.jar",
-    sha256 = "f198967436a5e7a69cfd182902adcfbcb9f2e41b349e1a5c8881a2407f615962",
+    url = "http://oss.sonatype.org/content/groups/public/org/scalatest/scalatest_2.12/3.0.3/scalatest_2.12-3.0.3.jar",
+    sha256 = "353f7c2bdde22c4286ee6a3ae0e425a9463b102f4c4cf76055a24f4666996762",
+  )
+
+  # scalatest has macros, note http_jar is invoking ijar
```

Comment:
I don't understand how the comment and the code fit together. Doesn't the ijar mess things up? I hope to send late next week a PR to add `scala_import` which will disable ijar (maybe with an option to turn it on)

Reply:
This code predates me and I don't understand how all pieces fit together, either.

Reply:
ijar destroys macros because it removes all code attributes from the jars. So if a dependency has macros it cannot filter through ijar.

One approach to solve this, is as ittaiz said, just build a `scala_import` that always (or by default) doesn't use ijar.

I don't know how what you have here can actually be working since it seems like you are stripping the macros, but maybe it isn't working yet.